cli/client/dcoscli.go

```diff
@@ -11,42 +10,48 @@ import (
 
 // TODO(nick): Consider breaking this config retrieval out into a separate independent library?
 
+// RunCLICommand is used to run generic commands against the DC/OS CLI.
+// It attempts to run the `dcos` executable contained within the user's PATH.
 func RunCLICommand(arg ...string) (string, error) {
 	if config.Verbose {
-		log.Printf("Running DC/OS CLI command: dcos %s", strings.Join(arg, " "))
+		PrintMessage("Running DC/OS CLI command: dcos %s", strings.Join(arg, " "))
 	}
 	outBytes, err := exec.Command("dcos", arg...).CombinedOutput()
 	if err != nil {
 		execErr, ok := err.(*exec.Error)
 		if ok && execErr.Err == exec.ErrNotFound {
 			// special case: 'dcos' not in PATH. provide special instructions
-			log.Printf("Unable to run DC/OS CLI command: dcos %s", strings.Join(arg, " "))
-			log.Printf("Please perform one of the following fixes, then try again:")
-			log.Printf("- Update your PATH environment to include the path to the 'dcos' executable, or...")
-			log.Printf("- Move the 'dcos' executable into a directory listed in your current PATH (see below).")
-			log.Fatalf("Current PATH is: %s", os.Getenv("PATH"))
+			PrintMessage("Unable to run DC/OS CLI command: dcos %s", strings.Join(arg, " "))
+			PrintMessage("Please perform one of the following fixes, then try again:")
+			PrintMessage("- Update your PATH environment to include the path to the 'dcos' executable, or...")
+			PrintMessage("- Move the 'dcos' executable into a directory listed in your current PATH (see below).")
+			PrintMessageAndExit("Current PATH is: %s", os.Getenv("PATH"))
 		}
 		return string(outBytes), err
 	}
 	// trim any trailing newline (or any other whitespace) if present:
 	return strings.TrimSpace(string(outBytes)), nil
 }
 
+// RequiredCLIConfigValue retrieves the CLI configuration property for name. If no value can
+// be retrieved, this terminates with a fatal error.
 func RequiredCLIConfigValue(name string, description string, errorInstruction string) string {
 	output, err := RunCLICommand("config", "show", name)
 	if err != nil {
-		log.Printf("Unable to retrieve configuration value %s (%s) from CLI. %s:",
+		PrintMessage("Unable to retrieve configuration value %s (%s) from CLI. %s:",
 			name, description, errorInstruction)
-		log.Printf("Error: %s", err.Error())
-		log.Printf("Output: %s", output)
+		PrintMessage("Error: %s", err.Error())
+		PrintMessage("Output: %s", output)
 	}
 	if len(output) == 0 {
-		log.Fatalf("CLI configuration value %s (%s) is missing/unset. %s",
+		PrintMessageAndExit("CLI configuration value %s (%s) is missing/unset. %s",
 			name, description, errorInstruction)
 	}
 	return output
 }
 
+// OptionalCLIConfigValue retrieves the CLI configuration for name. If no value can
+// be retrieved, this returns an empty string.
 func OptionalCLIConfigValue(name string) string {
```

Comment:
This is used in commands.go in the New method. I think it has to stay public for now.